### PR TITLE
sign up api's work complete and navigate to create shop

### PR DESCRIPTION
**Pull request by dassajib**
Sign up work done using react-query, zustand. Please review the code and give me feedback bhai.

### CODE REVIEW

File: src/Components/Shops/CreateShop.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { Button, Form, Input, Typography } from 'antd';
+import { CiShop } from 'react-icons/ci';
+
+const CreateShop = () => {
+  return (
```
Comment (salman1204) [inline src/Components/Shops/CreateShop.tsx]: use 'useForm' hook to initialise the Form. and add the Datatype there which will help you to validate the dataType of the form. 

`const [form] = useForm<FormSubmitEventInterface>()`

File: src/Components/SignUp/SignUp.tsx [hunk 2]
```diff
@@ -1,16 +1,45 @@
...
 import React from 'react';
 import { CiUser } from 'react-icons/ci';
 import { IoIosPhonePortrait } from 'react-icons/io';
 import { TbPassword } from 'react-icons/tb';
 import { Link } from 'react-router-dom';
+import { useSignupStore } from '../../store/useSignUpStore';
+import { validatePasswords } from '../../utils/validation';
+import { useSignUp } from '../../hooks/useSignUp';
 
 const SignUp: React.FC = () => {
```
Comment (salman1204) [inline src/Components/SignUp/SignUp.tsx]: React.FC is unnecessary . Currently the new version of react doesn't need this.

File: src/index.css [hunk 3]
```diff
@@ -6,23 +6,26 @@
...
+/* signin/signup */
+.signin,
+.signup,
+.create-shop {
   display: flex;
   justify-content: center;
   align-items: center;
   height: 100vh;
   width: 100vw;
 }
```
Comment (salman1204) [inline src/index.css]: Don't put the local component's css class in the index.css file

File: src/interface/index.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+export interface SignupData {
```
Comment (salman1204) [inline src/interface/index.ts]: make separate file for putting signUp interface. Don't put all the interface of the entire application in a single file. Categorise it and make separate file according that.

**Comment by salman1204** [comment]
@dassajib UI is not up to the mark. also the signup card gets increase while removing the phone number. The cards are too narrow for the bigger screen. 

> Make sure you are utilising Typescript properly.
> Do study on antd Form as you have to work a lot with Form. 
> Try to use tailwind class As much as possible rather than writing css by yourself.
> Pay Attention to the UI and take care of responsiveness.
> Validate the Data of the form item. 
> Before sending any PR, make sure you have written your best code.But don't try to be perfectionist all the time, it will kill your efficiency. Just be balanced. 

thanks.I appreciate your efforts.

File: src/main.tsx [hunk 5]
```diff
@@ -3,11 +3,16 @@ import ReactDOM from 'react-dom/client';
...
 ReactDOM.createRoot(document.getElementById('root')!).render(
   <React.StrictMode>
-    <BrowserRouter>
-      <App />
-    </BrowserRouter>
+    <QueryClientProvider client={queryClinet}>
+      <BrowserRouter>
+        <App />
+      </BrowserRouter>
+    </QueryClientProvider>
```
Comment (salman1204) [inline src/main.tsx:17]: Why you call the BrowserRouter twice? I think, you should not wrap the application with BrowserRouter twice. Instead, you should structure it in a way where the QueryClientProvider and BrowserRouter are properly nested

something like below : 
` <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </QueryClientProvider>`

File: .env [hunk 6]
```diff
@@ -0,0 +1 @@
+VITE_BACKEND_URL="http://127.0.0.1:8000/api/v1"
```
Comment (salman1204) [inline .env]: Don't put .env in the git. Add this file in the git ignore.

And create a new file named  .env.example where you just write the variables name of the .env file. not the values of those.

File: src/interface/index.ts (hunk 4 above)
Comment (salman1204) [reply src/interface/index.ts]: Also follow a naming pattern for interface, Ex: SignupDataInterface,SignupResponseInterface

File: src/main.tsx (hunk 5 above)
Comment (dassajib) [reply src/main.tsx:17]: bhai i can not understand where the problem is. My current code of main.tsx is matched with your example that you gave.

File: src/Components/Shops/CreateShop.tsx (hunk 1 above)
Comment (salman1204) [reply src/Components/Shops/CreateShop.tsx]: I don't see any changes.Can you confirm this, you have changed according to my comment?

File: .env (hunk 6 above)
Comment (salman1204) [reply .env]: delete this .env file from this branch. There won't be any file called .env

File: src/Components/Shops/CreateShop.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    <div className="flex justify-center items-center w-screen h-screen">
+      <Form
+        name="create-shop"
+        style={{
+          backgroundColor: 'rgba(255, 255, 255, 0.33)',
+          boxShadow: '0px 8px 32px 0 rgba(0, 0, 255, 0.33)',
+          padding: '60px 60px 60px 60px',
+          borderRadius: '10px',
+        }}
```
Comment (salman1204) [inline src/Components/Shops/CreateShop.tsx]: If possible, handle it with tailwind css.
may be something like below: 
`className="bg-white/30 shadow-[0_8px_32px_rgba(0,0,255,0.33)] p-16 rounded-lg"`

**Review by salman1204 — CHANGES_REQUESTED** [review]
Add a "Page" folder from where we will add index file for each route.

**Review by salman1204 — APPROVED** [review]

**Pull request closed without merging**